File: typescript/examples/open_orders.ts
#!/usr/bin/env npx ts-node
// @ts-nocheck
/**
 * Open Orders Example
 *
 * View all open orders with details.
 *
 * Requires: PRIVATE_KEY environment variable
 */

import { HyperliquidSDK } from '@quicknode/hyperliquid-sdk';

const PRIVATE_KEY = process.env.PRIVATE_KEY;

if (!PRIVATE_KEY) {
  console.error("Set PRIVATE_KEY environment variable");
  console.error("Example: export PRIVATE_KEY='0x...'");
  process.exit(1);
}

async function main() {
  const sdk = new HyperliquidSDK(undefined, { privateKey: PRIVATE_KEY });

  // Get all open orders
  const result = await sdk.openOrders();
  console.log(`Open orders: ${result.count}`);

  for (const o of result.orders) {
    const side = o.side === "B" ? "BUY" : "SELL";
    console.log(`  ${o.name} ${side} ${o.sz} @ ${o.limitPx} (OID: ${o.oid})`);
  }

  // Get order status for a specific order
  // const status = await sdk.orderStatus(12345);
  // console.log(`Order status: ${JSON.stringify(status)}`);
}

main().catch(console.error);

File: python/examples/open_orders.py
#!/usr/bin/env python3
"""
Open Orders Example

View all open orders with details.

Requirements:
    pip install hyperliquid-sdk

Usage:
    export PRIVATE_KEY="0x..."
    python open_orders.py
"""

import os
import sys

from hyperliquid_sdk import HyperliquidSDK

PRIVATE_KEY = os.environ.get("PRIVATE_KEY")

if not PRIVATE_KEY:
    print("Error: Set PRIVATE_KEY environment variable")
    print("  export PRIVATE_KEY='0xYourPrivateKey'")
    sys.exit(1)


def main():
    print("Hyperliquid Open Orders Example")
    print("=" * 50)

    sdk = HyperliquidSDK(private_key=PRIVATE_KEY)
    print(f"Address: {sdk.address}")

    # Get all open orders
    result = sdk.open_orders()
    print(f"Open orders: {result.count}")

    for o in result.orders:
        side = "BUY" if o.side == "B" else "SELL"
        print(f"  {o.name} {side} {o.sz} @ {o.limit_px} (OID: {o.oid})")

    # Get order status for a specific order
    # status = sdk.order_status(12345)
    # print(f"Order status: {status}")


if __name__ == "__main__":
    main()
